### PR TITLE
tests: Simplify tests

### PR DESCRIPTION
**Pull request by kumaraditya303**
Notable changes:

- [x] Use focal by default for CI
- [x] Test on Python 3.9
- [x] Convert reference counting script to a test
- [x] Convert installation test script to a test and test for all platforms not only for linux

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM ubuntu:focal
 
 # 1. Install latest Python
-RUN apt-get update && apt-get install -y python3 python3-pip && \
+RUN apt-get update && apt-get install -y python3 python3-pip unzip && \
```
Comment (kumaraditya303) [inline Dockerfile:4]: `unzip` is required to fix wheel's `RECORD` file

File: .github/workflows/test_docker.yml [hunk 2]
```diff
@@ -34,9 +34,10 @@ jobs:
     - name: Test
       run: |
         CONTAINER_ID="$(docker run --rm -v $(pwd):/root/playwright --name playwright-docker-test -d -t playwright-python:localbuild /bin/bash)"
+        docker exec --workdir /root/playwright/ "${CONTAINER_ID}" apt-get install unzip python3-venv -y
```
Comment (kumaraditya303) [inline .github/workflows/test_docker.yml:37]: `unzip` is required for updating wheel's `RECORD` file and `python3-venv` is required to run `virtualenv` tests

File: Dockerfile [hunk 3]
```diff
@@ -57,7 +57,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     xvfb
 
 # 8. Feature-parity with node.js base images.
-RUN apt-get update && apt-get install -y --no-install-recommends git ssh
+RUN apt-get update && apt-get install -y --no-install-recommends git ssh unzip
```
Comment (kumaraditya303) [inline Dockerfile:60]: `unzip` is required for fixing `RECORD` file of wheel

**Comment by kumaraditya303** [comment]
ping @pavelfeldman PR is ready for review

File: tests/test_generation_scripts.py [hunk 4]
```diff
@@ -1,29 +1,39 @@
...
 
 import pytest
 
-CAN_RUN_GENERATION_SCRIPT = sys.version_info >= (3, 8)
-
-if CAN_RUN_GENERATION_SCRIPT:
-    from scripts.generate_async_api import main as generate_async_api
-    from scripts.generate_sync_api import main as generate_sync_api
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 8), reason="requires python3.8 or higher"
```
Comment (pavelfeldman) [inline tests/test_generation_scripts.py:22]: It now requires 3.9 or higher

**Review by pavelfeldman — COMMENTED** [review]
Looks great!

File: tests/test_generation_scripts.py (hunk 4 above)
Comment (kumaraditya303) [reply tests/test_generation_scripts.py]: Fixed

**Comment by kumaraditya303** [comment]
@pavelfeldman PR is rebased and ready to be merged

**Review by pavelfeldman — APPROVED** [review]

**Comment by pavelfeldman** [comment]
There is a conflict...

**Comment by kumaraditya303** [comment]
@pavelfeldman fixed now, merge ASAP 🚀

**Comment by kumaraditya303** [comment]
ping @pavelfeldman

**Comment by kumaraditya303** [comment]
@pavelfeldman can you get this PR merge aa the conflict is resolved

**Comment by kumaraditya303** [comment]
@pavelfeldman conflict is resolved

**Comment by kumaraditya303** [comment]
@mxschmitt I have rebased and, PR was already approved by @pavelfeldman so can you review this and merge ?

**Review by mxschmitt — APPROVED** [review]
thats cool.

**Comment by kumaraditya303** [comment]
Seems they will pass but can wait sure

**Pull request merged**